CMSgov/mint-app
# [EASI-3012] [EASI-3522] Migrate Go LaunchDarkly SDK from 5 to 6

# EASI-3012 EASI-3522

## Changes and Description

- Upgrade Go SDK for LaunchDarkly to a version that is no longer EOL.
- Followed [this guide](https://docs.launchdarkly.com/sdk/server-side/go/migration-5-to-6)

Note: I did _NOT_ upgrade from 5 to 7, instead opting to go from 5 to 6 in order to minimize the number of breaking changes in the PR. The purpose of this ticket was mainly to get off the version of the SDK going EOL, and this accomplishes that

## How to test this change

- Ensure you have your LD env vars set in `.envrc.local`
- Start the app locally, signing in with your EUA
- Using LD, flip feature flags and ensure that both the Frontend and the Backend honor those flags
  - A quick check on the FE could be downgrading your user type and ensuring the home page updates to match
  - A quick check on the BE could be downgrading your user type (from Assessment) and ensuring that discussions don't show up with the ⭐ next to them

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
